docs/examples/reflections/plot_fit_iam_models.py

```diff
@@ -0,0 +1,98 @@
+
+"""
+IAM Model Fitting
+================================
+
+Illustrates how to fit an IAM model to data using :py:func:`~pvlib.iam.fit`.
+
+"""
+
+# %%
+# An incidence angle modifier (IAM) model quantifies the fraction of direct
+# irradiance is that is reflected away from a module's surface. Three popular
+# IAM models are Martin-Ruiz :py:func:`~pvlib.iam.martin_ruiz`, physical
+# :py:func:`~pvlib.iam.physical`, and ASHRAE :py:func:`~pvlib.iam.ashrae`.
+# Each model requires one or more parameters.
+#
+# Here, we show how to use
+# :py:func:`~pvlib.iam.fit` to estimate a model's parameters from data.
+#
+# Model fitting require a weight function that can assign
+# more influence to some AOI values than others. We illustrate how to provide
+# a custom weight function to :py:func:`~pvlib.iam.fit`.
+
+import numpy as np
+from random import uniform
+import matplotlib.pyplot as plt
+
+from pvlib.tools import cosd
+from pvlib.iam import (martin_ruiz, physical, fit)
+
+
+# %%
+# Fitting an IAM model to data
+# ----------------------------
+#
+# Here, we'll show how to fit an IAM model to data.
+# We'll generate some data by perturbing output from the Martin-Ruiz model to
+# mimic measured data and then we'll fit the physical model to the perturbed
+# data.
+
+# Create and perturb IAM data.
+aoi = np.linspace(0, 90, 100)
```

Comment:
```suggestion
aoi = np.linspace(0, 85, 18)
```
This would more representative for measurements.